lib/requester/request-wrapper.js

```diff
@@ -103,21 +144,47 @@ var _ = require('lodash'),
         var request = options.request,
             isSSL = _.startsWith(request.url.protocol, 'https');
 
-        if (!isSSL || !(options.certificateManager && options.certificateManager.getCertificateContents)) {
+        if (!isSSL) {
             return cb(null, options);
         }
 
-        // Get any certificates for this request from the certificate manager
-        options.certificateManager.getCertificateContents(request.url.getRemote(),
-            function (err, info) {
-                _.assign(options, {
+        if (_.get(options, 'certificateManager.getCertificateContents')) {
```

Comment:
Refactored to give more clarity on control. 

It might still look a bit complex, but will become cleaner once `certificateManager` is removed in the next breaking release.